Modify unit file to automatically restart the service on failure

jsf9k opened the pull request:
## 🗣 Description ##

This pull request modifies the unit file for `vncserver@1.service` so that the service is automatically restarted when the service fails.

## 💭 Motivation and context ##

Sometimes BurpSuitePro runs out of memory and is killed by the kernel, bringing down this service in the process.  This change doesn't fix that issue, but it does automatically restart the service so that the users can continue on without asking dev folks to intervene.

Closes #19.

## 🧪 Testing ##

All `pre-commit` hooks and `molecule` tests pass.  I also created a new Kali AMI for staging with these changes and verified that it functioned as expected:  see cisagov/kali-packer#66 for more details.

## ✅ Checklist ##

* [x] This PR has an informative and human-readable title.
* [x] Changes are limited to a single goal - _eschew scope creep!_
* [x] All relevant type-of-change labels have been added.
* [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
* [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
* [x] All new and existing tests pass.

jsf9k committed:
Modify the unit file to automatically restart the service on failure

Sometimes BurpSuitePro runs out of memory and is killed by the kernel,
bringing down this service in the process.  This change doesn't fix
that issue, but it does automatically restart the service so that the
users can continue on without asking dev folks to intervene.

This commit addresses #19.

dav3r reviewed (approved):
👍 ⚙️ 👍 

Is there anything in place that will prevent this service from continually restarting if it fails each time it starts up?  It's probably not a huge concern here, but wanted to ask just in case.

jsf9k commented:
>   
> 
> Is there anything in place that will prevent this service from continually restarting if it fails each time it starts up? It's probably not a huge concern here, but wanted to ask just in case.

There are system-level default values for `StartLimitIntervalSec` and `StartLimitBurst` in `/etc/systemd/system.conf`.  Looking now, though, it seems those defaults don't help in this case:
```console
$ grep -i defaultstart /etc/systemd/system.conf
#DefaultStartLimitIntervalSec=10s
#DefaultStartLimitBurst=5
```

See commit e03ef00.

jsf9k committed:
Set StartLimitIntervalSec and StartLimitBurst

The system-level defaults are not sufficient, given that we set
RestartSec to 30.

Co-authored-by: Dave Redmin <david.redmin@beta.dhs.gov>

jsf9k committed:
Add a validate command when creating the unit file

dav3r commented on tasks/configure_systemd.yml:
```diff
@@ -4,6 +4,7 @@
     src: vncserver@.service
     dest: /lib/systemd/system/
     mode: 0644
+    validate: /usr/bin/systemd-analyze validate %s
```
I didn't see anything in the docs for `systemd-analyze validate` - did you mean `verify`?
https://manpages.debian.org/buster/systemd/systemd-analyze.1.en.html
```suggestion
    validate: /usr/bin/systemd-analyze verify %s
```

jsf9k committed:
Fix typo in systemd-analyze command

Co-authored-by: Dave Redmin <david.redmin@beta.dhs.gov>

dav3r commented:
> > Is there anything in place that will prevent this service from continually restarting if it fails each time it starts up? It's probably not a huge concern here, but wanted to ask just in case.
> 
> There are system-level default values for `StartLimitIntervalSec` and `StartLimitBurst` in `/etc/systemd/system.conf`. Looking now, though, it seems those defaults don't help in this case:
> 
> ```
> $ grep -i defaultstart /etc/systemd/system.conf
> #DefaultStartLimitIntervalSec=10s
> #DefaultStartLimitBurst=5
> ```
> 
> See commit [e03ef00](https://github.com/cisagov/ansible-role-vnc-server/commit/e03ef00f633efe4ace97ec580bff6e726c08ff04).

That works for me- thanks!

jsf9k committed:
Comment out systemd-analyze validation

It would be great to use systemd-analyze verify to verify the VNC unit
file syntax, but that won't work.  See, for instance, this GitHub
issue comment:
https://github.com/ansible/ansible/issues/19243#issuecomment-266508074

mcdonnnj commented on tasks/configure_systemd.yml:
```diff
@@ -4,6 +4,10 @@
     src: vncserver@.service
     dest: /lib/systemd/system/
     mode: 0644
+    # It would be great to use systemd-analyze verify here, but that
```
😟

mcdonnnj commented on templates/vncserver@.service:
```diff
@@ -52,6 +55,13 @@ ExecStartPre=/bin/sh -c "/usr/bin/vncserver -kill :%i > /dev/null 2>&1 || :"
+# Sometimes BurpSuitePro runs out of memory and is killed by the
+# kernel, bringing down this service in the process.  This doesn't fix
+# that issue, but it does automatically restart the service so that
+# the assessors can continue on without asking dev folks to intervene.
```
Just a thought, but is there a more catch-all term to use instead of "dev folks" as we will hopefully move away from care and feeding of environments?

jsf9k replied:
How about this?
```suggestion
# the assessors can continue on without needing anyone else to intervene.
```

mcdonnnj replied:
Perfect!

mcdonnnj reviewed (approved):
Makes since. Thanks for digging into the test failures and the excellent documentation.

jsf9k committed:
Tweak language in comment

At some point the dev folks will not be maintaining the system, so the language here should be more generic.
